ipa-core/src/protocol/prss/mod.rs

```diff
@@ -75,17 +76,32 @@ impl Debug for UsedSet {
 /// This is public so that it can be used by the instrumentation wrappers in
 /// `ipa_core::protocol::context`.  It should not generally be used outside the PRSS implementation.
 #[derive(Clone, Copy, PartialEq, Eq)]
-pub(crate) struct PrssIndex128(u128);
+pub(crate) struct PrssIndex128 {
+    index: PrssIndex,
+    offset: u32,
+}
 
-impl From<u128> for PrssIndex128 {
-    fn from(value: u128) -> Self {
-        Self(value)
+impl Display for PrssIndex128 {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}:{}", self.index.0, self.offset)
+    }
+}
+
+#[cfg(debug_assertions)]
+impl From<u64> for PrssIndex128 {
+    fn from(value: u64) -> Self {
+        Self {
+            index: u32::try_from((value >> 32) & u64::from(u32::MAX))
```

Comment:
value is unsigned so >> must be a logical shift. I am missing something apparently what is the reason for doing `& u32::MAX`?

Reply:
I just wrote the two cases (this and the one on the next line) the same way, even though masking isn't necessary in this one. I may have been rewriting it in a hurry to appease clippy.